spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -335,6 +347,21 @@ private Dataset<String> listedFileDS() {
     return spark().createDataset(completeMatchingFileRDD.rdd(), Encoders.STRING());
   }
 
+  private String dataLocation() {
```

Comment:
If we go this route, I think it would be good to use the `LocationProvider` to get the data and metadata location.

Reply:
Currently `org.apache.iceberg.io.LocationProvider` has APIs to only operate on data location. Also APIs needs an input in fileName.
Is the suggestion to enhance the LocationProviders to include more APIs for data and metadata location?

Reply:
`LocationUtil.stripTrailingSlash(table.locationProvider().newDataLocation(""))` can do, but it's not the prettiest and also we need one for `write.metadata.location`.

Reply:
+1, @Fokko this shortcoming also come up in this PR, https://github.com/apache/iceberg/pull/11906#discussion_r1929534587